src/App.tsx

```diff
@@ -1,24 +1,240 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import * as postService from './api/todos';
+import { getFilteredTodos } from './utils/functions';
+import { TodoItem } from './components/TodoItem';
+import { TodoFilter } from './components/TodoFilter';
+import { Header } from './components/Header';
+import { ErrorMessage } from './components/ErrorMessage';
+import { TempTodo } from './components/TempTodo';
+import { Status } from './types/Status';
+import { ErrorMessages } from './utils/ErrorMessages';
 
-const USER_ID = 0;
+const USER_ID = 11457;
+const initialTodos: Todo[] = [];
 
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[]>(initialTodos);
+  const [filterBy, setFilterBy] = useState(Status.All);
+  const [errorMessage, setErrorMessage] = useState(ErrorMessages.NoError);
+  const [title, setTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingId, setLoadingId] = useState<number[]>([]);
+  const [isLoaderActive, setIsLoaderActive] = useState(false);
+  const [isToggled, setIsToggled] = useState(false);
+
+  const filteredTodos = getFilteredTodos(todoList, filterBy);
+
+  useEffect(() => {
+    postService.getTodos(USER_ID)
+      .then(setTodoList)
+      .catch(() => {
+        setErrorMessage(ErrorMessages.LoadError);
+      });
+    const timerId = setInterval(() => {
+      setErrorMessage(ErrorMessages.NoError);
+    }, 3000);
+
+    return () => {
+      clearInterval(timerId);
+    };
+  }, []);
+
+  const allTodosId = todoList.map(({ id }) => id);
+
+  const completedTodosId = todoList
+    .filter(({ completed }) => completed === true)
+    .map(({ id }) => id);
```

Comment:
You can do the same thing through `reduce` and walk through the array once instead of twice, which will improve the performance of your code.